hr_holidays_accrual/models/hr_leave_allocation.py

```diff
@@ -0,0 +1,412 @@
+                        _('Loss due to accrued amount limit')
+                    ))
+                    number_of_days += loss
+
+            date_from += period
+
+        if (self.limit_accumulated_days
+                and number_of_days > self.max_accumulated_days):
+            loss = self.max_accumulated_days - number_of_days
+            accruements.append((
+                loss,
+                date_to.date(),
+                _('Loss due to accumulation limit')
+            ))
+            number_of_days += loss
+
+        _logger.info(
+            '%s days of "%s" leave allocated to employee "%s"',
+            number_of_days,
+            self.holiday_status_id.name,
+            self.employee_id.name,
+        )
+
+        return accruements, number_of_days
+
+    def _get_accrual_period(self):
+        if self.interval_unit == 'weeks':
+            return relativedelta(weeks=self.interval_number)
+        if self.interval_unit == 'months':
+            return relativedelta(months=self.interval_number)
+        if self.interval_unit == 'years':
+            return relativedelta(years=self.interval_number)
+
+    def _get_date_from(self):
+        if self.date_from:
+            return self.date_from
+
+        service_start_date = self.employee_id.sudo().service_start_date
+        if service_start_date:
+            return datetime.combine(
+                service_start_date,
+                datetime.min.time()
+            )
+
+        return self.employee_id.sudo().create_date
+
+    def _get_date_to(self):
+        if self.date_to:
+            return self.date_to
+
+        service_termination_date = (
+            self.employee_id.sudo().service_termination_date
+        )
+        if service_termination_date:
+            return datetime.combine(
+                service_termination_date,
+                datetime.min.time()
+            )
+
+        return None
+
+    def _get_days_to_accrue(
+        self,
+        period_start,
+        period,
+        period_end,
+        as_of_datetime,
+        days_worked,
+        workable_days,
+    ):
+        days_to_accrue = self.number_per_interval
+        if self.unit_per_interval == 'hours':
+            days_to_accrue /= (
+                self.employee_id.resource_calendar_id.hours_per_day
+            ) or HOURS_PER_DAY
+
+        if (self.accrual_method == 'period_start'
+                and period_start < as_of_datetime):
+            return (
+                days_to_accrue,
+                period_start.date(),
+                _('Start-of-period accruement')
+            )
+        elif (self.accrual_method == 'period_end'
+                and period_start + period < as_of_datetime):
+            return (
+                days_to_accrue,
+                period_end.date(),
+                _('End-of-period accruement')
+            )
+        elif self.accrual_method == 'prorate' and workable_days > 0:
+            return (
```

Comment:
For all those return values, could you use a NamedTuple ? Then you wouldn't need to access each tuple properties with their position, but can use a name to explicit what's happening.